packages/@lwc/engine-server/src/__tests__/fixtures.spec.ts

```diff
@@ -109,7 +109,11 @@ function formatHTML(src: string): string {
         }
 
         if (start !== pos) {
-            res += getPadding() + src.slice(start, pos) + '\n';
+            // Ignore the zero-width space character used to mark empty text nodes.
+            const content = src.slice(start, pos).replace(/\u200D/g, '');
+            if (content.length > 0) {
+                res += getPadding() + content + '\n';
+            }
```

Comment:
Can we move this into the `engine-server` serialization logic itself? It seems odd for SSR to emit this character if we don't need to.